### PR TITLE
feat(server): Implement `CLIENT KILL`

### LINKED ISSUE #1614: Add support for CLIENT KILL command

### PR DESCRIPTION
**Pull request by chakaz**
Currently, it supports the following syntax:

* `CLIENT KILL <addr>:<port>`
* `CLIENT KILL ID <id>`
* `CLIENT KILL ADDR <addr>:<port>`
* `CLIENT KILL LADDR <addr>:<port>`

It will not allow killing an admin-connection from a non-admin port.

There are a few parameters of `CLIENT KILL` that Redis supports but this PR does not yet add. Let's add them as needed.

Fixes #1614
Related to #1495

### CODE REVIEW

**Review by kostasrim — COMMENTED** [review]
lgtm, maybe add a test?

**Comment by chakaz** [comment]
> lgtm, maybe add a test?

great idea! PTAL :)

File: tests/dragonfly/server_family_test.py [hunk 1]
```diff
@@ -74,15 +74,24 @@ async def test_get_databases(async_client: aioredis.Redis):
...
+            assert len(await client_conn.execute_command("CLIENT LIST")) == 2
+            assert len(await admin_client.execute_command("CLIENT LIST")) == 2
+
+            # Can't kill admin from regular connection
+            with pytest.raises(Exception) as e_info:
+                await client_conn.execute_command("CLIENT KILL LADDR 127.0.0.1:1112")
+
+            await admin_client.execute_command("CLIENT KILL LADDR 127.0.0.1:1111")
+            with pytest.raises(Exception) as e_info:
+                await client_conn.ping()
```
Comment (adiholden) [inline tests/dragonfly/server_family_test.py:94]: maybe add client list command call after the kill

File: src/server/server_family.cc [hunk 2]
```diff
@@ -506,6 +506,70 @@ void ClientTracking(CmdArgList args, ConnectionContext* cntx) {
...
+  for (auto* listener : listeners) {
+    listener->TraverseConnections(cb);
+  }
+
+  if (kill_errors.load() == 0) {
+    return cntx->SendLong(killed_connections.load());
+  } else {
+    return cntx->SendError(absl::StrCat("Killed ", killed_connections.load(),
+                                        " client(s), but unable to kill ", kill_errors.load(),
+                                        " client(s) from admin port."));
```
Comment (adiholden) [inline src/server/server_family.cc:569]: why do you write "from admin port"? its not only admin port that kills the connections

Comment (chakaz) [reply src/server/server_family.cc]: better now?

**Review by adiholden — APPROVED** [review]

**Pull request merged**